### PR TITLE
Update UG to specify prefixes constraints for add and delete commands

### LINKED ISSUE #229: [PE-D] Stipend accepts null
No value is provided to stipend and the software accepts it. However, in UG, it is stated that the stipend should accept a positive integer up to 10 digits long.

[Screen Recording 2022-04-01 at 2.56.16 PM.mov](https://raw.githubusercontent.com/punnyhuimin/ped/main/files/74de8a59-9f6e-4625-ace2-b742d4191345.mov)

-------------
Labels: `type.DocumentationBug` `severity.High`
original: punnyhuimin/ped#7

### PR DESCRIPTION
**Pull request by BrendonLau**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -417,18 +422,21 @@ not, `editCompany` has got you covered. Simply edit the parts of the company det
...
 * Existing values will be updated to the input values.
+* The `PHONE_NUMBER`, `EMAIL` and `ADDRESS` here can be left blank, allowing you to remove previously added information which may be erroneous.
 * More restrictions for command parameters can be found [here](#c-company-restrictions)
 
 **Examples:**
 
 * `list` followed by `editCompany 1 p/91234567 e/johndoe@example.com`edits the 1<sup>st</sup> company in the displayed
   company list.
+* `list` followed by `editCompany 1 p/ e/`removes `PHONE_NUMBER` and `EMAIL` from the 1<sup>st</sup> company in the displayed
+  company list.
```
Comment (kerbysoh) [inline docs/UserGuide.md]: Should this be part of the same line?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -408,6 +408,11 @@ company that may come in handy during the internship application process.
 
 </div>
 
+<div markdown="block" class="alert alert-danger">
+:warning: If a prefix is specified, even for optional fields, the parameter cannot be blank. e.g. `p/` only is not valid but `p/94442222` is valid.
```
Comment (kerbysoh) [inline docs/UserGuide.md]: Small suggestion here to use "value after the prefix" rather than parameter.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -544,6 +552,7 @@ company.
 <div markdown="block" class="alert alert-danger">
 
 :warning: When round brackets are used, they must be in pairs.
+:warning: If a prefix is specified, even for optional fields, the parameter cannot be blank. e.g. `$/` only is not valid but `$/1000` is valid.
```
Comment (kerbysoh) [inline docs/UserGuide.md]: We should have a separate div here to signify different warnings.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -555,7 +564,7 @@ Just got invited to an interview? Keep your internship role status and other det
 
 **Function:** Edits the specified role in the specified company within the displayed company list.
 
-**Format:** `editRole COMPANY_INDEX ROLE_INDEX [n/ROLE_NAME [(TYPE)]] [s/STATUS] [r/REMINDER_DATE] [d/DESCRIPTION] [$/STIPEND]`
+**Format:** `editRole COMPANY_INDEX ROLE_INDEX n/ROLE_NAME [(TYPE)] s/STATUS [r/REMINDER_DATE] [d/DESCRIPTION] [$/STIPEND]`
```
Comment (kerbysoh) [inline docs/UserGuide.md]: If we are changing the meaning of the brackets here, it must be updated above in the command format to specify what exactly the brackets mean.

**Review by kerbysoh — COMMENTED** [review]
Good work! Take a look at some of my minor suggestions

File: docs/UserGuide.md (hunk 4 above)
Comment (BrendonLau) [reply docs/UserGuide.md]: you are right i added them back.

File: docs/UserGuide.md (hunk 3 above)
Comment (BrendonLau) [reply docs/UserGuide.md]: ok

File: docs/UserGuide.md (hunk 2 above)
Comment (BrendonLau) [reply docs/UserGuide.md]: yup changed. thanks

File: docs/UserGuide.md (hunk 1 above)
Comment (BrendonLau) [reply docs/UserGuide.md:439]: what you mean? like `list` followed by `editCompany 1 p/91234567 e/johndoe@example.com`edits the 1<sup>st</sup> company in the displayed

Comment (BrendonLau) [reply docs/UserGuide.md]: what you mean? like * `list` followed by `editCompany 1 p/91234567 e/`edits the PHONE_NUMBER and removes EMAIL from the company in the displayed list?

**Pull request merged**